xbmc/addons/interfaces/Addon/AddonCallbacksAddon.cpp

```diff
@@ -102,21 +101,21 @@ void CAddonCallbacksAddon::AddOnLog(void *addonData, const addon_log_t addonLogL
 
   try
   {
-    int xbmcLogLevel = LOGNONE;
+    int xbmcLogLevel;
     switch (addonLogLevel)
     {
-      case LOG_ERROR:
-        xbmcLogLevel = LOGERROR;
-        break;
+      case LOG_DEBUG:
+        xbmcLogLevel = LOGDEBUG;
       case LOG_INFO:
         xbmcLogLevel = LOGINFO;
-        break;
-      case LOG_NOTICE:
-        xbmcLogLevel = LOGNOTICE;
-        break;
-      case LOG_DEBUG:
+      case LOG_WARNING:
+        xbmcLogLevel = LOGWARNING;
+      case LOG_ERROR:
+        xbmcLogLevel = LOGERROR;
+      case LOG_FATAL:
+        xbmcLogLevel = LOGFATAL;
       default:
-        xbmcLogLevel = LOGDEBUG;
+        xbmcLogLevel = LOGNONE;
```

Comment:
Maybe add a comment the line before saying:
`// If the log level is invalid then don't log anything`

Reply:
Then in the future we know what this log level is for!